framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -82,22 +86,217 @@ describe('CommitPool', () => {
+
+			when(commitPool['_chain'].dataAccess.getBlockHeaderByHeight as any)
+				.calledWith(maxHeightCertified)
+				.mockResolvedValue({ aggregateCommit: { height: maxHeightCertified } } as never);
+
+			commitPool['_bftAPI'].getBFTHeights = jest.fn().mockResolvedValue({ maxHeightPrecommitted });
+			commitPool['_bftAPI'].getCurrentValidators = jest
+				.fn()
+				.mockResolvedValue(Array.from({ length: 103 }, () => getRandomBytes(32)));
+		});
+
+		it('should clean all the commits from nonGossipedCommit list with height below removal height', async () => {
+			// Assert
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(6);
+			// Arrange
+			commitPool['_bftAPI'].existBFTParameters = jest.fn().mockResolvedValue(true);
+			const context = createNewAPIContext(new InMemoryKVStore());
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits and stale commit is deleted
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(0);
+		});
+
+		it('should clean all the commits from gossipedCommit list with height below removal height', async () => {
+			// Assert
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(6);
+			// Arrange
+			commitPool['_bftAPI'].existBFTParameters = jest.fn().mockResolvedValue(true);
+			const context = createNewAPIContext(new InMemoryKVStore());
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(10);
+		});
+
+		it('should clean all the commits from nonGossipedCommit that does not have bftParams change', async () => {
+			commitPool['_nonGossipedCommits'].set(1070, [
+				{
+					blockID: getRandomBytes(32),
+					certificateSignature: getRandomBytes(96),
+					height: 1070,
+					validatorAddress: getRandomBytes(20),
+				},
+			]);
+			// Assert
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(7);
+			// Arrange
+			const bftParamsMock = jest.fn();
+			commitPool['_bftAPI'].existBFTParameters = bftParamsMock;
+			const context = createNewAPIContext(new InMemoryKVStore());
+			when(bftParamsMock).calledWith(context, 1071).mockResolvedValue(false);
+			when(bftParamsMock).calledWith(context, maxHeightCertified).mockResolvedValue(true);
+			when(bftParamsMock)
+				.calledWith(context, height + 1)
+				.mockResolvedValue(true);
+			// Act
+			await commitPool['_job'](context);
+			// Assert
+			// nonGossiped commits are moved to gossiped commits
+			expect([...commitPool['_nonGossipedCommits'].values()].flat()).toHaveLength(0);
+			expect([...commitPool['_gossipedCommits'].values()].flat()).toHaveLength(10);
+			expect(commitPool['_nonGossipedCommits'].get(1070)).toBeUndefined();
+		});
+
+		it('should clean all the commits from gossipedCommit that does not have bftParams change', async () => {
```

Comment:
I think this test is redundant (same arrange, act, assert) and can be merged with the previous (L202)

Reply:
they are different, the previous is adding failing BFT param commit to nonGossiped and this one in gossiped and then later we are checking for commits presence in the list it was added. But code seems very redundant so I have merged both the tests in one